frontend/storage/utils/fetchGameData.ts

```diff
@@ -0,0 +1,75 @@
+import { GameSeries } from '../models/GameSeries';
+import { Game } from '../models/Game';
+import { ApiRequests } from './apiRequestManager';
+import { GameEnvironment } from '../models/GameEnvironment';
+import { Environment } from '../models/Environment';
+import { DefaultImage, Image } from '../models/Image';
+import { AbilityExample } from '../models/AbilityExample';
+import { Ability } from '../models/Ability';
+import { AbilityGroup } from '../models/AbilityGroup';
+import { AbilityCategory } from '../models/AbilityCategory';
+import { createGameAbilitiesListData } from './createGameAbilitiesListData';
+
+export async function fetchGameViewData(id: number) {
+	try {
+		const game: Game = await ApiRequests.get(`games/${id}`, {});
+		const series: GameSeries = await ApiRequests.get(`game-series/${game.seriesId}`, {});
+		return { game, series };
+	} catch (error) {
+		throw new Error(error);
+	}
+}
+export async function fetchGameBoxData(gameId: number, imageId: number | undefined) {
+	try {
+		const gameEnvironments: Array<GameEnvironment> = await ApiRequests.get(`game-environments`, { gameId: gameId });
+		let image: Image;
+		if (imageId) {
+			image = await ApiRequests.get(`images/${imageId}`, {});
+		} else {
+			image = DefaultImage;
+		}
+		const environments: Array<Environment> = new Array();
+		for (const gameEnvironment of gameEnvironments) {
+			environments.push(await ApiRequests.get('environments/' + gameEnvironment.environmentId, {}));
+		}
+		return { image, environments };
+	} catch (error) {
+		throw new Error(error);
+	}
+}
+export async function fetchGameAbilitiesListData(gameId: number) {
```

Comment:
I might as well tell you this already, the ultimate goal I want to reach with the CRs on this PR will be set of functions for the endpoints, but preparing of the props will be handled by `Smart<component>`s. In order words all this logic you crammed here would be extracted out. But let's take it one step at a time!